Content.Server/Physics/Controllers/ChasingWalkSystem.cs

```diff
@@ -0,0 +1,98 @@
+using System.Linq;
+using Robust.Shared.Random;
+using Robust.Shared.Timing;
+
+using Content.Server.Physics.Components;
+using Content.Shared.Follower.Components;
+using Content.Shared.Throwing;
+using Robust.Shared.Physics.Systems;
+using Robust.Shared.Physics.Components;
+
+namespace Content.Server.Physics.Controllers;
+
+/// <summary>
+/// A system which makes its entity chasing another entity with selected component.
+/// </summary>
+internal sealed class ChasingWalkSystem : EntitySystem
+{
+
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly IGameTiming _gameTiming = default!;
+    [Dependency] private readonly EntityLookupSystem _lookup = default!;
+    [Dependency] private readonly SharedPhysicsSystem _physics = default!;
+
+
+    private EntityQuery<TransformComponent> _xformQuery;
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        _xformQuery = GetEntityQuery<TransformComponent>();
+    }
```

Comment:
none of this is needed, Transform() has an internal transform query

replace it with a PhysicsComponent query to use later